### PR TITLE
Add post mortem for incident on the 6th February 2023

### PR DESCRIPTION
**Pull request by krauselukas**

### CODE REVIEW

**Review by hennevogel — CHANGES_REQUESTED** [review]

File: _posts/deployments/2023-02-06-post-mortem.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+Feburary 3th, 2023 around 11:37PM until February 6th, 2023 13:06PM.
+
+## Root Causes
+
+The problem was caused by failing mailer jobs, which started to block the delayed job queue. The mailer jobs were failing
+due to not properly handling hidden projects.
+
+## Trigger
+
+A recently setup hidden project on our [reference server](https://build.opensuse.org) server.
```
[the remark this answers is not in the document]
Comment (hennevogel) [reply _posts/deployments/2023-02-06-post-mortem.md:23]: something happened on Friday around 22:30 to `multimedia:libs:cisco-openh264`. What?

File: _posts/deployments/2023-02-06-post-mortem.md [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+## Impact
+
+Our [reference server](https://build.opensuse.org) wasn't sending out email notifications (or only partially) to the users starting
+Feburary 3th, 2023 around 11:37PM until February 6th, 2023 13:06PM.
+
+
+## Root Causes
+
+The problem was caused by failing mailer jobs, which started to block the delayed job queue. The mailer jobs were failing
```
[the remark this answers is not in the document]
Comment (krauselukas) [reply _posts/deployments/2023-02-06-post-mortem.md]: I adapted the text, please have another look :)

File: _posts/deployments/2023-02-06-post-mortem.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (krauselukas) [reply _posts/deployments/2023-02-06-post-mortem.md]: I adapted the text, please have another look :)

File: _posts/deployments/2023-02-06-post-mortem.md [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+## Impact
+
+Our [reference server](https://build.opensuse.org) wasn't sending out email notifications (or only partially) to the users starting
+Feburary 3th, 2023 around 11:37PM until February 6th, 2023 13:06PM.
+
+## Root Causes
+
+The problem was caused by an exception thrown in the `SendEventEmailsJob` due to not properly querying "hidden projects".
+Since those jobs were failing, the delayed job queue started to be filled with unprocessable jobs. Those were continuously
+failing and blocked other mailer jobs.
```
Comment (hennevogel) [inline _posts/deployments/2023-02-06-post-mortem.md:20]: Still not true. Nothing blocked other mailer jobs. There is just one mail job and it crashed :)

File: _posts/deployments/2023-02-06-post-mortem.md [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+## Trigger
+
+A recent change of user roles on a "hidden project" on our [reference server](https://build.opensuse.org) server, started
+to trigger queuing `SendEventEmailsJob`'s to send emails to the subscribed users.
+
+## Resolution
+
+Temporarily ignoring the events coming from the hidden project in the `SendEventEmailsJob` in order to allow queuing
+and processing of all other mailer jobs again.
```
Comment (hennevogel) [inline _posts/deployments/2023-02-06-post-mortem.md:30]:
```suggestion
and processing of mailer jobs again.
```

File: _posts/deployments/2023-02-06-post-mortem.md [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+Collaboration among the team to resolve the issue.
+
+### What went wrong?
+
+Not considering the default scope when querying the projects in the `SendEventEmailsJob`.
+
+### Where we got lucky?
+
+Once the problem was clear, it was easy to temporarily exclude the project that caused the issue from the `SendEventEmailsJob`.
+Fortunately there is only one project affected, that had to be excluded for now.
```
Comment (hennevogel) [inline _posts/deployments/2023-02-06-post-mortem.md:57]: I don't think this is info the internet needs to know 😺

**Review by hennevogel — CHANGES_REQUESTED** [review]

File: _posts/deployments/2023-02-06-post-mortem.md (hunk 5 above)
Comment (krauselukas) [reply _posts/deployments/2023-02-06-post-mortem.md]: true, removed

File: _posts/deployments/2023-02-06-post-mortem.md [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+
+## Timeline (CEST)
```
[the remark this answers is not in the document]
Comment (krauselukas) [reply _posts/deployments/2023-02-06-post-mortem.md]: upps

**Review by hennevogel — APPROVED** [review]

**Pull request merged**